web/yo/app/scripts/controllers/genes.js

```diff
@@ -132,32 +126,97 @@ angular.module('oncokbApp')
                             $scope.metaFlags[genes[i]].review = true;
                             flag = false;
-                            if (curationQueueChecked) {
-                                break;
-                            }
-                        }
-                        if (uuids[j] === 'CurationQueueArticles') {
-                            $scope.metaFlags[genes[i]].CurationQueueArticles = geneMetaData.get('CurationQueueArticles');
-                            curationQueueChecked = true;
+                            break;
                         }
                     }
                     if (flag) {
                         $scope.metaFlags[genes[i]].review = false;
                     }
+                    if (geneMetaData.get('CurationQueueArticles') > 0) {
+                        $scope.metaFlags[genes[i]].CurationQueueArticles = geneMetaData.get('CurationQueueArticles');
+                    } else {
+                        $scope.metaFlags[genes[i]].CurationQueueArticles = 0;
+                    }
+                    if (geneMetaData.get('AllArticles') > 0) {
+                        genesWithArticle.push(genes[i]);
+                    }
+                }
+                $scope.documents = Documents.get();
+                $scope.status.rendering = false;
+                var geneNames = [];
+                for (var i = 0; i < $scope.documents.length; i++) {
+                    geneNames.push($scope.documents[i].title);
+                }
+                $scope.geneNames = geneNames;
+                if (genesWithArticle.length > 0) {
+                    $scope.searchCuration(genesWithArticle, 0);
+                } else {
+                    $scope.status.queueRendering = false;
+                    $scope.queue = [];
                 }
             }
-
+            var curationResults = [];
+            $scope.searchCuration = function(genesForCuration, index) {
+                var documents = Documents.get({title: genesForCuration[index]});
+                var document = _.isArray(documents) && documents.length === 1 ? documents[0] : null;
+                if (document) {
+                    storage.getRealtimeDocument(document.id).then(function(realtime) {
+                        if (realtime && realtime.error) {
+                            dialogs.error('Error', 'Fail to load ' + genesForCuration[index] + ' document. Please contact the developer.');
+                        } else {
+                            var model = realtime.getModel();
+                            var queueModel = model.getRoot().get('queue');
+                            if (queueModel) {
+                                _.each(queueModel.asArray(), function(item) {
+                                    if (item.get('curator') && MainUtils.isExpiredCuration(item.get('dueDay')) && !item.get('notified')) {
+                                        item.set('notified', new Date().getTime());
```

Comment:
If the curation has expired, we will send the second email automatically when they enter the page. The email is sent in secondTimeAutoNotify() in curationQueue.js. After we sent the email, we need to mark it in both queue and queueModel. In gene page, it is straight forward. But in genes page, marking 'notified' in queue is done is secondTimeAutoNotify(), while marking 'notified' in queueModel is done in searchCuration() in genes.js. The reason is to avoid fetching the same list of queueModels repeatedly.